internal/parser/planparserv2/parser_visitor.go

```diff
@@ -1605,3 +1712,396 @@ func (v *ParserVisitor) VisitTemplateVariable(ctx *parser.TemplateVariableContex
+		"L2":     true, // Euclidean distance
+		"IP":     true, // Inner product distance
+		"COSINE": true, // Cosine distance
+		// More supported metric types can be added as needed
+		"HAMMING": true, // Hamming distance (for binary vectors)
+		"JACCARD": true, // Jaccard distance (for binary vectors)
+	}
+
+	// Support case insensitive
+	return validMetrics[strings.ToUpper(metricType)]
+}
+
+// VisitAliasExpression implements AS expressions like expr AS alias
+func (v *ParserVisitor) VisitAliasExpression(ctx *parser.AliasExpressionContext) interface{} {
+	// Constants definition
+	const (
+		errContextNil   = "alias expression context is nil"
+		errNoExpression = "no expression found in alias expression"
+		errNoIdentifier = "no identifier found in alias expression"
+		errInvalidExpr  = "invalid expression in alias expression"
+		errEmptyAlias   = "alias cannot be empty"
+		errInvalidAlias = "invalid alias format"
+	)
+
+	// Enhanced input validation
+	if ctx == nil {
+		return fmt.Errorf(errContextNil)
+	}
+
+	// Validate expression exists
+	if ctx.Expr() == nil {
+		return fmt.Errorf(errNoExpression)
+	}
+
+	// Validate identifier exists
+	if ctx.Identifier() == nil {
+		return fmt.Errorf(errNoIdentifier)
+	}
+
+	// Get expression
+	exprResult := ctx.Expr().Accept(v)
+	if exprResult == nil {
+		return fmt.Errorf(errInvalidExpr)
+	}
+
+	if err := getError(exprResult); err != nil {
+		return fmt.Errorf("expression error in alias: %w", err)
+	}
+
+	expr, ok := exprResult.(*ExprWithType)
+	if !ok || expr == nil {
+		return fmt.Errorf(errInvalidExpr)
+	}
+
+	// Get alias and validate
+	alias := ctx.Identifier().GetText()
+	if alias == "" {
+		return fmt.Errorf(errEmptyAlias)
+	}
+
+	// Validate alias format (letters, numbers, underscores)
+	if !isValidAliasFormat(alias) {
+		return fmt.Errorf("%s: %s", errInvalidAlias, alias)
+	}
+
+	// Create aliased expression - using new AliasedExpr protocol
+	return &ExprWithType{
+		expr: &planpb.Expr{
+			Expr: &planpb.Expr_AliasedExpr{
+				AliasedExpr: &planpb.AliasedExpr{
+					Expr:  expr.expr,
+					Alias: alias,
+				},
+			},
+		},
+		dataType: expr.dataType,
+	}
+}
+
+// isValidAliasFormat validates if alias format is valid
+func isValidAliasFormat(alias string) bool {
+	if len(alias) == 0 || len(alias) > 64 {
+		return false
+	}
+
+	// Must start with letter or underscore
+	first := alias[0]
+	if !((first >= 'a' && first <= 'z') ||
+		(first >= 'A' && first <= 'Z') ||
+		first == '_') {
+		return false
+	}
+
+	// Remaining characters must be letters, numbers or underscores
+	for i := 1; i < len(alias); i++ {
+		char := alias[i]
+		if !((char >= 'a' && char <= 'z') ||
+			(char >= 'A' && char <= 'Z') ||
+			(char >= '0' && char <= '9') ||
+			char == '_') {
+			return false
+		}
+	}
+
+	return true
+}
+
+// isLikelyDistanceQueryMisuse 智能判断是否可能是距离查询的误用
+// Based on context analysis, not hardcoded aliases
+func isLikelyDistanceQueryMisuse(aliasName, fieldName string) bool {
```

Comment:
don't think this really make sense.

The key is to make sure this expr only for output field expr instead of put it into filter expression